### PR TITLE
Feature/12 custom users table

### PR DESCRIPTION
**Pull request by xperseguers**

### CODE REVIEW

File: Classes/Hook/UserSettings.php [hunk 1]
```diff
@@ -65,6 +71,13 @@ class UserSettings
     /*─────────────────────────────────────────────────────────────────────────────*\
             Methods
     \*─────────────────────────────────────────────────────────────────────────────*/
+    public function __construct()
+    {
+        // Note: Dependency Injection does not work in this context
+        $this->objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        $this->dispatcher = GeneralUtility::makeInstance(Dispatcher::class);
```
Comment (codeFareith) [inline Classes/Hook/UserSettings.php:78]: Codacy Issue found: [Avoid using static access to class '\TYPO3\CMS\Core\Utility\GeneralUtility' in method '__construct'.](https://app.codacy.com/manual/robinvonberg/cf_google_authenticator/pullRequest?prid=5687502)

**Comment by codeFareith** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/robinvonberg/cf_google_authenticator/pullRequest?prid=5687502&bid=18616743)

File: Classes/Hook/UserSettings.php [hunk 2]
```diff
@@ -65,6 +71,13 @@ class UserSettings
     /*─────────────────────────────────────────────────────────────────────────────*\
             Methods
     \*─────────────────────────────────────────────────────────────────────────────*/
+    public function __construct()
+    {
+        // Note: Dependency Injection does not work in this context
+        $this->objectManager = GeneralUtility::makeInstance(ObjectManager::class);
```
Comment (codeFareith) [inline Classes/Hook/UserSettings.php:77]: Codacy Issue found: [Avoid using static access to class '\TYPO3\CMS\Core\Utility\GeneralUtility' in method '__construct'.](https://app.codacy.com/manual/robinvonberg/cf_google_authenticator/pullRequest?prid=5687502)

Comment (xperseguers) [reply Classes/Hook/UserSettings.php:77]: so you prefer to use another trait marked as deprecated and to be removed in version 1.2.0 but still in use at various places anyway?

**Comment by codeFareith** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/robinvonberg/cf_google_authenticator/pullRequest?prid=5687502&bid=18616743)

File: Classes/Hook/UserSettings.php [hunk 3]
```diff
@@ -152,7 +152,19 @@ private function getLayer(): string
             $layer = 'Backend';
         }
 
-        return $layer;
+        $dispatcher = GeneralUtility::makeInstance(Dispatcher::class);
```
Comment (codeFareith) [inline Classes/Hook/UserSettings.php:155]: Codacy Issue found: [Avoid using static access to class '\TYPO3\CMS\Core\Utility\GeneralUtility' in method 'getLayer'.](https://app.codacy.com/manual/robinvonberg/cf_google_authenticator/pullRequest?prid=5687502)

**Pull request merged**